code/game/turfs/flooring/flooring.dm

```diff
@@ -35,52 +38,217 @@ var/list/flooring_types
+	footstep_sound = "grass"
+
+/decl/flooring/dirt
+	name = "dirt"
+	desc = "Do they smoke grass out in space, Bowie? Or do they smoke AstroTurf?"
+	icon = 'icons/turf/flooring/dirt.dmi'
+	icon_base = "dirt"
+	flags = TURF_REMOVE_SHOVEL | SMOOTH_ONLY_WITH_ITSELF
+	build_type = null //Todo: add bags of fertilised soil or something to create dirt floors
+	footstep_sound = "gravel"
 
 /decl/flooring/asteroid
 	name = "coarse sand"
 	desc = "Gritty and unpleasant."
 	icon = 'icons/turf/flooring/asteroid.dmi'
 	icon_base = "asteroid"
-	flags = TURF_HAS_EDGES | TURF_REMOVE_SHOVEL
+	flags = TURF_HAS_EDGES | TURF_REMOVE_SHOVEL | TURF_CAN_BURN | TURF_CAN_BREAK
 	build_type = null
+	footstep_sound = "asteroid"
+
+
 
+
+
+
+//=========PLATING==========
 /decl/flooring/reinforced/plating
 	name = "plating"
-	icon = 'icons/turf/floors.dmi'
+	descriptor = "plating"
+	icon = 'icons/turf/flooring/plating.dmi'
 	icon_base = "plating"
-	build_type = null
-	flags = TURF_HAS_EDGES | TURF_HAS_CORNERS | SMOOTH_ONLY_WITH_ITSELF
+	build_type = /obj/item/stack/material/steel
+	flags = TURF_HAS_EDGES | TURF_HAS_CORNERS | TURF_REMOVE_WELDER | TURF_CAN_BURN | TURF_CAN_BREAK
 	can_paint = 1
-
-/decl/flooring/hull
+	plating_type = /decl/flooring/reinforced/plating/under
+	is_plating = TRUE
+	footstep_sound = "plating"
+	removal_time = 150
+	health = 100
+
+//Normal plating allows anything, except other types of plating
+/decl/flooring/reinforced/plating/can_build_floor(var/decl/flooring/newfloor)
+	if (istype(newfloor, /decl/flooring/reinforced/plating))
+		return FALSE
+	return TRUE
+
+/decl/flooring/reinforced/plating/get_plating_type(var/turf/location)
+	if (turf_is_upper_hull(location))
+		return null
+	return plating_type
+
+
+
+
+//==========UNDERPLATING==============
+/decl/flooring/reinforced/plating/under
+	name = "underplating"
+	icon = 'icons/turf/flooring/plating.dmi'
+	descriptor = "support beams"
+	icon_base = "under"
+	build_type = /obj/item/stack/material/steel //Same type as the normal plating, we'll use can_build_floor to control it
+	flags = TURF_REMOVE_WRENCH | TURF_CAN_BURN | TURF_CAN_BREAK
+	can_paint = 1
+	plating_type = /decl/flooring/reinforced/plating/hull
+	is_plating = TRUE
+	removal_time = 250
+	health = 200
+	resistance = 12
+	footstep_sound = "catwalk"
+
+//Underplating can only be upgraded to normal plating
+/decl/flooring/reinforced/plating/under/can_build_floor(var/decl/flooring/newfloor)
+	if (newfloor.type == /decl/flooring/reinforced/plating)
+		return TRUE
+	return FALSE
+
+/decl/flooring/reinforced/plating/under/attackby(var/obj/item/I, var/mob/user, var/turf/T)
+	if (istype(I, /obj/item/stack/rods))
+		.=TRUE
+		var/obj/item/stack/rods/R = I
+		if(R.amount <= 3)
+			return
+		else
+			R.use(3)
+			user << SPAN_NOTICE("You start connecting [R.name]s to [src.name], creating catwalk ...")
+			if(do_after(user,60))
+				T.alpha = 0
+				var/obj/structure/catwalk/CT = new /obj/structure/catwalk(T)
+				T.contents += CT
```

Comment:
That line is weird and redundant.

Reply:
it is, but its harmless. just a leftover thing i forgot to remove